contracts/core/Allo.sol

```diff
@@ -1,79 +1,359 @@
+    /// @dev 1e18 = 100%, 1e17 = 10%, 1e16 = 1%, 1e15 = 0.1%
+    uint256 public feePercentage;
+
+    /// @notice Incremental index
+    uint256 private _poolIndex;
+
+    /// @notice Nonce to create salt for clone strategy
+    uint256 private _nonce;
+
+    /// @notice Allo treasury
+    address payable public treasury;
+
+    /// @notice Registry of pool creators
+    Registry public registry;
+
+    /// @notice Pool.id -> Pool
+    mapping(uint256 => Pool) public pools;
+
+    /// @notice Strategy -> bool
+    mapping(address => bool) public approvedStrategies;
+
+    /// @notice Strategy -> bool
+    mapping(address => bool) public usedStrategies;
+
+    /// ======================
+    /// ======= Events =======
+    /// ======================
+
+    event PoolCreated(
+        uint256 indexed poolId,
+        bytes32 indexed identityId,
+        address allocationStrategy,
+        address distributionStrategy,
+        address token,
+        uint256 amount,
+        Metadata metadata
+    );
+
+    event PoolMetadataUpdated(uint256 indexed poolId, Metadata metadata);
+
+    event PoolFunded(uint256 indexed poolId, uint256 amount, uint256 fee);
+
+    event PoolClosed(uint256 indexed poolId);
+
+    event TreasuryUpdated(address treasury);
+
+    event FeeUpdated(uint256 fee);
+
+    event RegistryUpdated(address registry);
+
+    /// ====================================
+    /// =========== Intializer =============
+    /// ====================================
+
+    /// @notice Initializes the contract after an upgrade
+    /// @dev During upgrade -> an higher version should be passed to reinitializer
+    /// @param _registry The address of the registry
+    /// @param _treasury The address of the treasury
+    /// @param _feePercentage The fee percentage
+    function initialize(address _registry, address payable _treasury, uint256 _feePercentage) public reinitializer(1) {
+        _initializeOwner(msg.sender);
+
+        registry = Registry(_registry);
+        treasury = _treasury;
+        feePercentage = _feePercentage;
+
+        emit RegistryUpdated(_registry);
+        emit TreasuryUpdated(_treasury);
+        emit FeeUpdated(_feePercentage);
+
+        __Multicall_init();
+    }
+
+    /// ====================================
+    /// =========== Modifier ===============
+    /// ====================================
+
+    modifier isPoolAdmin(uint256 _poolId) {
+        if (!registry.isOwnerOrMemberOfIdentity(pools[_poolId].identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+        _;
+    }
+
+    /// ====================================
+    /// ==== External/Public Functions =====
+    /// ====================================
+
+    /// @notice Returns the pool info
+    /// @param _poolId The id of the pool
+    /// @dev calls out to the registry to get the identity metadata
+    function getPoolInfo(uint256 _poolId) external view returns (Pool memory) {
+        return pools[_poolId];
+    }
```

Comment:
I don't think this is needed, since we have the `pools` public getter.

Reply:
Agreed, we can remove this.